buildSrc/build.gradle.kts

```diff
@@ -6,8 +8,23 @@ repositories {
     gradlePluginPortal()
 }
 
+// TODO define versions in Gradle Version Catalog https://github.com/Kotlin/dokka/pull/2884
+val properties = file("../gradle.properties").inputStream().use {
+    Properties().apply { load(it) }
+}
+
+val kotlinVersion = properties["kotlin_version"]
+
 dependencies {
-    implementation("com.github.jengelman.gradle.plugins:shadow:2.0.4")
-    implementation("org.jetbrains.kotlinx:binary-compatibility-validator:0.10.1")
+    // Import Gradle Plugins that will be used in the buildSrc pre-compiled script plugins, and any `build.gradle.kts`
+    // files in the project.
+    // Use their Maven coordinates (plus versions), not Gradle plugin IDs!
+    // This should be the only place that Gradle plugin versions are defined, so they are aligned across all build scripts
+
+    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
+    implementation("gradle.plugin.com.github.johnrengelman:shadow:7.1.2")
+    implementation("org.jetbrains.kotlinx:binary-compatibility-validator:0.12.1")
```

Comment:
```suggestion
    implementation("org.jetbrains.kotlinx:binary-compatibility-validator:0.13.0")
```

Reply:
I'm not opposed to bumping this, but I want to try and limit this PR as much as possible. I think it would be better to update in a separate PR.